### PR TITLE
Fix label wrap (#5612) and wrap at ZERO WIDTH SPACE (#5613)

### LINKED ISSUE #5612: msLayoutTextSymbol replaces WRAP char unconditionaly with SPACE
The "new" msLayoutTextSymbol function replaces *all* occurences of the `WRAP` character with a SPACE character and then wraps labels at these characters if they are longer then `MAXLENGTH`. For example, if you use `WRAP '-'` the label `My-label-with-dash` is rendered as `My label with dash` even if the label is not wrapped.

This appears to be a regression with MapServer 7. (/cc @tbonfort)

Is there a reason for this behavior? My understanding is that we can remove:

https://github.com/mapserver/mapserver/blob/2b9d02203e38ecaad1214ef463ea501749824432/textlayout.c#L534-L540

and replace the checks for `' '` with `ts->label->wrap` at these positions:

https://github.com/mapserver/mapserver/blob/2b9d02203e38ecaad1214ef463ea501749824432/textlayout.c#L543
https://github.com/mapserver/mapserver/blob/2b9d02203e38ecaad1214ef463ea501749824432/textlayout.c#L549

I can provide a patch, but please consider my follow up issue.

### PR DESCRIPTION
**Pull request by olt**
This fixes #5612 and implements #5613.

Here are two images from the added test case.

This is the output with current master. Lines starting with W should wrap, lines with NW should not wrap. Note how the dash is removed in the first lines and how the second line is wrapped even if there is no wrap character. (#5612)

labels-wrap-master

This is the output with my fix. The first two lines are wrapped correctly. The lower lines show how it is possible to wrap at arbitrary positions by inserting ZERO WIDTH SPACE characters.
labels-wrap-fixed

Questions:
- I've added a test. How do I generate the expected image? Most "result images perceptually match, though files differ" on my system.
- The patch is against master but I can rebase it. Where? branch-7-2? branch-7-3?

I'll update to the documentation.

### CODE REVIEW

File: textlayout.c [hunk 1]
```diff
@@ -531,27 +530,17 @@ int msLayoutTextSymbol(mapObj *map, textSymbolObj *ts, textPathObj *tgret) {
       }
     } else {
       assert(ts->label->maxlength > 0);
-      if(ts->label->wrap != ' '){
-        for(i=0;i<num_glyphs;i++) {
-          /* replace all occurences of the wrap character with a space */
-          if(glyphs.unicodes[i]== ts->label->wrap)
-            glyphs.unicodes[i]= ' ';
```
Comment (olt) [inline textlayout.c]: This is where wrap characters were replaced unconditionally with SPACE.

File: textlayout.c [hunk 2]
```diff
@@ -531,27 +530,17 @@ int msLayoutTextSymbol(mapObj *map, textSymbolObj *ts, textPathObj *tgret) {
...
-        }
-      }
       if(num_glyphs > ts->label->maxlength) {
+        int num_cur_glyph_on_line = 0; /*count for the number of glyphs on the current line*/
         for(i=0; i<num_glyphs; i++) {
-          if(glyphs.unicodes[i]== ' ')
-            break;
-        }
-        if(i != num_glyphs) { /* we have at least one space on which we can break */
```
Comment (olt) [inline textlayout.c]: Loop to check if we have a wrap character is not necessary, as the actual wrapping loop does not do any computation (except counting num_cur_glyph_on_line).

**Comment by sdlime** [comment]
Should we take this one to mapserver-dev?

**Comment by olt** [comment]
I wrote on mapserver-dev already, but I didn't ask for a vote: https://lists.osgeo.org/pipermail/mapserver-dev/2018-August/015464.html

**Comment by sdlime** [comment]
I saw that after my comment here. Let me reply to your note to try and solicit a little more review. If nothing then we’ll just move forward with the commit.
________________________________

**Comment by sdlime** [comment]
@olt - to confirm. No breaking changes on this one, correct? Seems appropriate for 7.0 or at worst 7.2 don't you think? --Steve

**Comment by olt** [comment]
No, unless fixing a regression is a breaking change ;-)
I'm not that familiar with the release schedule and versioning, to know if 7.0 or 7.2 is appropriate. Please let me know on which branch I should rebase this.

**Comment by sdlime** [comment]
Let’s go with 7.2.
________________________________

**Comment by olt** [comment]
Added the missing *expected* image and rebased on branch-7-2. Update to docs in https://github.com/mapserver/docs/pull/251

**Comment by sdlime** [comment]
Thanks Oliver!

**Pull request merged**